cdmtaskservice/http_bearer.py

```diff
@@ -0,0 +1,63 @@
+"""
+Alteration of FastAPI's HTTPBearer class to handle the KBase authorization
+step and allow for more informative errors.
+
+Also adds an `optional` keyword argument that allows for missing, but not malformed,
+authentication headers. If `optional` is `True` and no authorization header is provided, `None`
+will be returned in place of the normal `KBaseUser`.
+"""
+
+from fastapi.security.http import HTTPBase
+from fastapi.openapi.models import HTTPBearer as HTTPBearerModel
+from fastapi.security.utils import get_authorization_scheme_param
+from fastapi.requests import Request
+from typing import Optional
+
+from cdmtaskservice import app_state
+from cdmtaskservice import kb_auth
+
+# Modified from https://github.com/tiangolo/fastapi/blob/e13df8ee79d11ad8e338026d99b1dcdcb2261c9f/fastapi/security/http.py#L100
+
+_SCHEME = "Bearer"
+
+
+class KBaseHTTPBearer(HTTPBase):
+    def __init__(
+        self,
+        *,
+        bearerFormat: Optional[str] = None,
+        scheme_name: Optional[str] = None,
+        description: Optional[str] = None,
+        # FastAPI uses auto_error, but that allows for malformed headers as well as just
+        # no header. Use a different variable name since the behavior is different.
+        optional: bool = False,
+        # Considered adding an admin auth role here and throwing an exception if the user
+        # doesn't have it, but often you want to customize the error message.
+        # Easier to handle that in the route method.
+    ):
```

Comment:
I am assuming there is no need to call `super().__init__`?

Reply:
The original source doesn't have an `__init__` method at all, so presumably not https://github.com/fastapi/fastapi/blob/e13df8ee79d11ad8e338026d99b1dcdcb2261c9f/fastapi/security/http.py#L100